### PR TITLE
chore: upgrade testrunner image

### PR DESCRIPTION
**Pull request by taegyunkim**
1. Upgrade testrunner image to the one generated by #10373 
2. testrunner now runs python 3.11.9 (upgraded from 3.11.3) and gevent isn't compatible with python 3.11.9, https://github.com/python/cpython/issues/117983, so update the test to run when version < 3.11.9
3. Regenerate riot file for appsec_integrations using python 3.12, as outdated pydantic doesn't work with 3.12.4, see https://github.com/pydantic/pydantic/issues/9637. This also resulted in updating pytest version and `setup_module` is used instead of `setup`.

## Checklist
- [x] PR author has checked that all the criteria below are met
- The PR description includes an overview of the change
- The PR description articulates the motivation for the change
- The change includes tests OR the PR description describes a testing strategy
- The PR description notes risks associated with the change, if any
- Newly-added code is easy to change
- The change follows the [library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html)
- The change includes or references documentation updates if necessary
- Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist
- [x] Reviewer has checked that all the criteria below are met 
- Title is accurate
- All changes are related to the pull request's stated goal
- Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes
- Testing strategy adequately addresses listed risks
- Newly-added code is easy to change
- Release note makes sense to a user of the library
- If necessary, author has acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment
- Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

### CODE REVIEW

**Review by brettlangdon — APPROVED** [review]

**Comment by taegyunkim** [comment]
Turned off auto-merge as it didn't trigger gitlab pipelines for some reason, and want to see gitlab pipelines running successfully.

**Review by avara1986 — APPROVED** [review]

File: hatch.toml
Comment (P403n1x87) [inline hatch.toml:1]: This looks like an unrelated change?

File: tests/appsec/integrations/test_psycopg2.py
Comment (P403n1x87) [inline tests/appsec/integrations/test_psycopg2.py:1]: Is this change related?

File: tests/profiling/collector/test_stack.py
Comment (P403n1x87) [inline tests/profiling/collector/test_stack.py:1]: Maybe this should go in a separate PR?

File: hatch.toml
Comment (taegyunkim) [reply hatch.toml:1]: Technically, it could go in a separate PR, but I noticed these tests suites running slow and could benefit from the env var.

File: tests/appsec/integrations/test_psycopg2.py
Comment (taegyunkim) [reply tests/appsec/integrations/test_psycopg2.py:1]: Yes, see the PR description.

File: tests/profiling/collector/test_stack.py
Comment (taegyunkim) [reply tests/profiling/collector/test_stack.py:1]: No, the profile circle ci fails without this change, as the new testrunner image upgrades python 3.11.3 from 3.11.9 which is not compatible with gevent.

File: tests/appsec/integrations/pygoat_tests/Dockerfile.pygoat.2.0.1
[the remark this answers is not in the document]
Comment (taegyunkim) [reply tests/appsec/integrations/pygoat_tests/Dockerfile.pygoat.2.0.1:1]: Well strictly speaking, this change is not directly related to this change. I noticed that appsec_integrations circle ci suites are taking very long time, more than 20 minutes, while working on this PR. This PR upgrades the testrunner image that is used everywhere in our ci pipelines. So, appsec_integrations were also run part of this PR and I was a surprised that they take that long. It takes ~7 minutes building this docker image and ~7 * 2 minutes running two riot venv hashes. Adding this env var reduces the build time marginally, and the more fundamental thing to do would be building this docker image in a separate pipeline and use tagged image instead of building as part of the ci. I have told asm folks to do that - https://dd.slack.com/archives/C02UE1NR5GC/p1724791798716899

**Review by P403n1x87 — APPROVED** [review]
🙌  thanks!
